underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystemFactory.java

```diff
@@ -33,39 +28,14 @@
 @ThreadSafe
 public final class HdfsUnderFileSystemFactory implements UnderFileSystemFactory {
   /**
-   * Cache mapping {@code Path}s to existing {@link UnderFileSystem} instances. The paths should be
-   * normalized to root paths because only their schemes and authorities are needed to identify
-   * which {@link FileSystem} they belong to.
-   */
-  private Map<Path, HdfsUnderFileSystem> mHdfsUfsCache = new HashMap<>();
-
-  /**
    * Constructs a new {@link HdfsUnderFileSystemFactory}.
    */
   public HdfsUnderFileSystemFactory() {}
 
   @Override
   public UnderFileSystem create(String path, Configuration configuration, Object conf) {
-    Preconditions.checkArgument(path != null, "path may not be null");
-
-    // Normalize the path to just its root. This is all that's needed to identify which FileSystem
-    // the Path belongs to.
-    Path rootPath = getRoot(new Path(path));
-    synchronized (mHdfsUfsCache) {
-      if (!mHdfsUfsCache.containsKey(rootPath)) {
-        mHdfsUfsCache
-            .put(rootPath, new HdfsUnderFileSystem(new AlluxioURI(path), configuration, conf));
-      }
-      return mHdfsUfsCache.get(rootPath);
-    }
-  }
-
-  private static Path getRoot(Path path) {
-    Path currPath = path;
-    while (currPath.getParent() != null) {
-      currPath = currPath.getParent();
-    }
-    return currPath;
+    Preconditions.checkNotNull(path);
```

Comment:
I think if we have a constant there should be minimal overhead. Things get bad if we start concatenating or using `String.format` in the argument since those values need to be evaluated before it can be passed to the Preconditions method.